tests/ui/generic-associated-types/issue-93340-1.rs

```diff
@@ -0,0 +1,19 @@
+// revisions: old next
+//[next] compile-flags: -Znext-solver
+pub trait Scalar: 'static {
+    type RefType<'a>: ScalarRef<'a>;
+}
+
+pub trait ScalarRef<'a>: 'a {}
+
+fn cmp_eq<'a, 'b, A: Scalar, B: Scalar, O: Scalar>(a: A::RefType<'a>, b: B::RefType<'b>) -> O {
+    todo!()
+}
+
+fn build_expression<A: Scalar, B: Scalar, O: Scalar>(
+) -> impl Fn(A::RefType<'_>, B::RefType<'_>) -> O {
+    cmp_eq
```

Comment:
This previously compiled because of incompleteness:
proving `for<'a, 'b> cmp_eq<?0, ?1, ?2>: Fn(A::RefType<'a>, B::RefType<'b>) -> O`

This constrained `?0` to `A` even though there could have been another type for which `<?x as Scalar>::RefType<'a> eq <A as Scalar>::RefType<'a>>` holds, this is related to https://github.com/rust-lang/rust/issues/102048 and is fixed because we now normalize `for<'a> <?x as Scalar>::RefType<'a>` after instantiating the binder, causing it to be replaced with a fresh infer var.

Reply:
I believe the (new) behavior is correct -- https://github.com/rust-lang/rust/pull/108918#issuecomment-1464102683.

I don't believe this test should have ever compiled because it *relies* on the incompleteness of not normalizing aliases with escaping bound regions.

Reply:
Yes, should have stated so explicitly, this breakage is desired imo